third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -315,17 +315,25 @@ ComputationClient::DataPtr PjRtComputationClient::ReplicateShardedData(
     XLA_CHECK_EQ(shards.size(), GetLocalDevices().size());
     std::vector<std::vector<ComputationClient::DataPtr>> arguments_by_device(
         GetLocalDevices().size(), std::vector<ComputationClient::DataPtr>(1));
-    for (auto shard : shards) {
+    for (int i = 0; i < shards.size(); ++i) {
```

Comment:
Can we reuse ShardingUtil::InputHandler here?

Reply:
+1 let's do that

Reply:
ah, forgot about the circular dependencies issue. We should try avoid using the xla sharding util in the runtime client.